sdk/go/common/util/gitutil/git.go

```diff
@@ -453,44 +593,58 @@ func parseHostAuth(u *url.URL) string {
 	return user + ":" + p + "@" + u.Host
 }
 
-// ParseGitRepoURL returns the URL to the Git repository and path from a raw URL.
-// For example, an input of "https://github.com/pulumi/templates/templates/javascript" returns
-// "https://github.com/pulumi/templates.git" and "templates/javascript".
-// Additionally, it supports nested git projects, as used by GitLab.
-// For example, "https://github.com/pulumi/platform-team/templates.git/templates/javascript"
-// returns "https://github.com/pulumi/platform-team/templates.git" and "templates/javascript"
-//
-// Note: URL with a hostname of `dev.azure.com`, are currently treated as a raw git clone url
-// and currently do not support subpaths.
-func ParseGitRepoURL(rawurl string) (string, string, error) {
+type gitRepoURLParts struct {
+	// URL is the base URL, without a path.
+	URL string
+	// Hostname is the actual hostname for the URL.
+	Hostname string
+	// Path is the path part of the URL, if any.
+	Path string
+}
+
+func parseGitRepoURLParts(rawurl string) (gitRepoURLParts, error) {
```

Comment:
solid improvement over "return N strings" 👍

Reply:
You can thank Joe for this one, it was in his original PR which I branched off of :)